src/java.base/share/conf/security/java.security

```diff
@@ -28,6 +28,45 @@
 # Properties in this file are typically parsed only once. If any of the
 # properties are modified, applications should be restarted to ensure the
 # changes are properly reflected.
+#
+# The special "include" property can be defined one or multiple times in
+# this file with a filesystem path value. The effect of each definition
+# is to include a referred security properties file inline, adding all
+# its properties and values. Included files, as well as files pointed by
+# java.security.properties, can include other files recursively. Paths
+# may be absolute or relative, and may contain system properties for
+# expansion in the form of ${system.property}. System properties in include
+# paths bring flexibility for users to switch between predefined security
+# profiles on a per-execution basis, as shown in one of the examples below.
+# If a system property does not have a value, it expands to the empty
+# string. This behavior enables the establishment of a default profile and
+# makes profile selection optional. Each relative path is resolved against
+# the base file containing its "include" definition, if local. An error will
+# be thrown if a file cannot be included, either because it does not exist,
+# cannot be resolved, or is recursively included more than once. For a
+# secure JDK configuration, it is important to check OS permissions assigned
+# to any file included.
+#
+# Examples:
+#
+#   include ${java.home}/conf/security/extra.security
+#
+#   include extra.security
+#
+#   include ${java.home}/conf/security/profile${SecurityProfile}.security
+#
+# Note: In the latter case, the SecurityProfile system property can be set to
+# select an existing security profile on a per-execution basis. For example,
+# if Java is launched with the argument -DSecurityProfile=FIPS, the profile
+# defined in the profileFIPS.security file is applied or an error thrown if
+# not found. Other files may exist in the same directory defining alternative
+# profiles. If SecurityProfile is not set, it expands to the empty string.
+# When a profile.security file exists (may be empty), it defines a default
+# security profile. When it does not exist, an error is thrown and explicit
+# profile selection is required. Security properties defined after an include
+# statement will override any matching property defined in a profile. In order
+# to avoid this behavior, the include statement may be placed at the end of
+# this file.
```

Comment:
It's probably better to place the newly added "Security properties defined after an include statement..." in an individual paragraph since it's not limited to "the latter case" (or you mean "last"?). Then you might not want to use the "profile" word since it's not defined elsewhere.

Reply:
Instead of "to avoid this behavior", how about "if you want the properties defined in the included file to override those in the main file"?

Reply:
We reorganized the documentation, split it into paragraphs, numbered examples and made some other changes. Changes include an explanation of what it means to "inline" a file in terms of overrides (for security properties defined both before and after the include statement). Please have a look.